http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -1292,4 +1320,43 @@ public RequestDistribution(HttpClient httpClient, Processor processor)
             });
         }
     }
+
+    // By wrapping HttpClient, we are able to substitute the underlying AuthenticationStore
+    // with a more efficient one.
+    private static class WrappedHttpClient
+            extends HttpClient
+    {
+        private final AuthenticationStore authenticationStore;
+        private final SpnegoAuthentication spnego;
+
+        public WrappedHttpClient(KerberosConfig kerberosConfig, HttpClientConfig config, HttpClientTransport transport, SslContextFactory sslContextFactory)
```

Comment:
Yeah, this class is just a hack to work around not being able to initialize jetty client with a custom auth store. I'll change the name, though.